core/network/endpoint_bindings.go

```diff
@@ -0,0 +1,27 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package network
+
+import "github.com/juju/errors"
+
+// MapWithSpaceNames returns the current bindingMap with space names rather than ids.
+func MapBindingsWithSpaceNames(bindingsMap map[string]string, lookup SpaceInfos) (map[string]string, error) {
```

Comment:
nit: would be nice to see a firmer contract for this func documented. Noting the fact lookup can be nil and the NotFound error.